src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -289,21 +294,31 @@ public function walkSelectorSource(QOM\SelectorInterface $source)
 
     /**
      * @param QOM\JoinInterface $source
+     * @param boolean $root
```

Comment:
can you please add an explanation of the semantics of the parameter? i know that this class is making a bad example with missing explanations, but we should work on improving the situation when we add new things.